packages/application-generic/src/services/auth/auth.service.ts

```diff
@@ -387,15 +384,35 @@ export class AuthService {
   private async getApiKeyUser({ apiKey }: { apiKey: string }): Promise<{
     environment?: EnvironmentEntity;
     user?: UserEntity;
-    key?: IApiKey;
     error?: string;
   }> {
-    const environment = await this.environmentRepository.findByApiKey(apiKey);
+    const hashedApiKey = createHash('sha256').update(apiKey).digest('hex');
+
+    /*
+     * backward compatibility - after encrypt-api-keys-migration execution:
+     * * update `findByApiKeyBackwardCompatibility` to `findByApiKey`
+     */
+    const environment =
+      await this.environmentRepository.findByApiKeyBackwardCompatibility({
+        key: apiKey,
+        hash: hashedApiKey,
+      });
+
     if (!environment) {
-      return { error: 'API Key not found' };
+      return { error: 'Environment not found' };
```

Comment:
This error text is returned to the API Client when their API key does not have a matching environment. Whilst this change is logical based on the code flow, it's not actually helpful to the API Client. Their API key may have had a simple spelling error, the proposed change might leave them scratching their head.
I'd suggest adding a comment here for future reference.
```suggestion
      // Failed to find the environment for the provided API key.
      return { error: 'API Key not found' };
```

Reply:
Good point, thanks for bringing it up.